### PR TITLE
File.deleteOnExit()

### PR DESCRIPTION
**Pull request by akashdw**
Addresses https://github.com/druid-io/druid/issues/3885

 * Removed deleteOnExit() from most of the classes except few(classes which heavily depend on this method).
-> ReplayableFirehose
-> SchemalessIndexTest
-> SpatialFilterBonusTest
-> SpatialFilterTest
-> DeterminePartitionsJobTest.java

 * Made IOpeon closable

### CODE REVIEW

File: processing/src/main/java/io/druid/segment/data/IOPeon.java [hunk 1]
```diff
@@ -19,15 +19,17 @@
...
 
 /**
  */
-public interface IOPeon
+public interface IOPeon extends Closeable
 {
   public OutputStream makeOutputStream(String filename) throws IOException;
   public InputStream makeInputStream(String filename) throws IOException;
+  @Deprecated
   public void cleanup() throws IOException;
```
Comment (b-slim) [inline processing/src/main/java/io/druid/segment/data/IOPeon.java]: is this method still used ? if not we can remove it thought ?

Comment (b-slim) [inline processing/src/main/java/io/druid/segment/data/IOPeon.java]: or why not just renaming it close ?

Comment (akashdw) [inline processing/src/main/java/io/druid/segment/data/IOPeon.java]: not used anywhere, Removing it.

**Comment by akashdw** [comment]
Thanks @b-slim. Addressed comments.

**Comment by b-slim** [comment]
👍

**Pull request merged**